velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp

```diff
@@ -595,3 +597,51 @@ TEST_F(ParquetReaderTest, parseLongTagged) {
   EXPECT_EQ(col0->type->kind(), TypeKind::BIGINT);
   EXPECT_EQ(type->childByName("_c0"), col0);
 }
+
+TEST_F(ParquetReaderTest, prefetchRowGroups) {
+  EXPECT_EQ(FLAGS_parquet_prefetch_rowgroups, 1);
+
+  auto rowType = ROW({"id"}, {BIGINT()});
+  const std::string sample(getExampleFilePath("multiple_row_groups.parquet"));
+
+  ReaderOptions readerOptions{defaultPool.get()};
+  // Disable preload of file
+  readerOptions.setFilePreloadThreshold(0);
+
+  ParquetReader reader = createReader(sample, readerOptions);
+  EXPECT_EQ(reader.numberOfRowGroups(), 4);
+
+  RowReaderOptions rowReaderOpts;
+  rowReaderOpts.setScanSpec(makeScanSpec(rowType));
+  auto rowReader = reader.createRowReader(rowReaderOpts);
+  auto parquetRowReader = dynamic_cast<ParquetRowReader*>(rowReader.get());
+
+  // The first two row groups should be prefetched.
+  EXPECT_TRUE(parquetRowReader->isRowGroupBuffered(0));
+  EXPECT_TRUE(parquetRowReader->isRowGroupBuffered(1));
+
+  // Read the first row group.
+  constexpr int kBatchSize = 1000;
+  auto result = BaseVector::create(rowType, kBatchSize, pool_.get());
+  auto actualRows = parquetRowReader->next(kBatchSize, result);
+  // kBatchSize should be large enough to hold the entire row group.
+  EXPECT_LE(actualRows, kBatchSize);
+
+  // Advance to the second row group.
+  parquetRowReader->nextRowNumber();
+  // Check the next row group has been prefetched.
```

Comment:
We can probably add `EXPECT_FALSE(parquetRowReader->isRowGroupBuffered(0));` here after fixing the check